cmd/metacache-manager.go

```diff
@@ -128,8 +128,8 @@ func (m *metacacheManager) getBucket(ctx context.Context, bucket string) *bucket
 	m.mu.Lock()
 	// See if someone else fetched it while we waited for the lock.
 	b, ok = m.buckets[bucket]
+	m.mu.Unlock()
```

Comment:
You introduce races when you don't hold the lock and you can easily have several goroutines load at the same time.

Revert this.

Reply:
what kind of race @klauspost ?

Reply:
My main attempt here is to not block the call for network calls, otherwise, the Lock() doesn't timeout and it's undefined.

Reply:
Either we have a deadline for such operations we have no way to timeout quickly.

Reply:
@harshavardhana Caches will be missing. 

You are welcome to add a timeout `loadBucketMetaCache` and change it to use the context. However it should not be derived from the caller, since we don't want client disconnects to affect loading.

Reply:
We want it to load or recreate the cache exactly once.